crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -6,13 +6,13 @@ use naga::valid::Capabilities;
 use naga::{valid::ModuleInfo, Module};
```

Comment:
i think it's useful functionality with or without #5703.

i don't see any problems replicating this into naga_oil. i would need to adjust the compiled module cache and header cache to key off def/value pairs instead of just defs, but i think otherwise it should be almost copy/paste.

i also don't see any real benefit to doing #5703 first.